app/assets/javascripts/index.js.erb

```diff
@@ -0,0 +1,666 @@
+				iconSize: [20, 20],
+				className: 'user-icon'
+			});
+
+			const marker = L.marker(latLng, {
+				icon,
+				riseOnHover: true
+			});
+			this.position = marker;
+
+			console.log(this.position);
+
+			this.map.addLayer(this.position);
+
+			console.log('Nice');
+		}
+		this.position.setLatLng(latLng);
+	}
+
+	/**
+	 * Get the current user position
+	 * @returns {L.LatLng} Position of the user
+	 */
+	getUserPosition() {
+		return this.position.getLatLng();
+	}
+
+	/**
+	 * Calculates the radius given the current map boundaries. This is critical for loading data.
+	 * @returns {number} Radius in KM
+	 */
+	getRadius() {
+		const bounds = this.map.getBounds();
+		const ne = bounds.getNorthEast();
+		const nw = bounds.getNorthWest();
+		const m = nw.distanceTo(ne);
+		const km = m / 1000;
+		const r = Math.min(Math.ceil(km / Math.sqrt(2)), MAX_RADIUS);
+
+		return r;
+	}
+}
+
+/**
+ * Wrapper of geolocation API
+ */
+class GPS {
+	/**
+	 * Creates a new GPS. Prevents us from having too many watchers.
+	 */
+	constructor() {
+		this.watchId = null;
+	}
+
+	/**
+	 * Fetches the current user location
+	 * @param {number} defaultLat - Default latitude
+	 * @param {number} defaultLng - Default longitude
+	 * @returns {Promise<L.LatLng>} A promise to the user's current location
+	 */
+	getCurrentPosition(defaultLat, defaultLng) {
+		const defaultPosition = L.latLng(defaultLat, defaultLng);
+
+		const options = {
+			enableHighAccuracy: true,
+			timeout: 5000,
+			maximumAge: 0
+		};
+
+		return new Promise(resolve => {
+			navigator.geolocation.getCurrentPosition(
+				position => {
+					const {
+						coords
+					} = position;
+					const {
+						latitude, longitude
+					} = coords;
+
+					resolve(L.latLng(latitude, longitude));
+				},
+				() => resolve(defaultPosition),
+				options
+			);
+		});
+	}
+
+	/**
+	 * Watches the current user's position and calls `onPosChange` when the user's position changes
+	 * @callback onPosChange
+	 * 		@param {L.LatLng} latLng - Contains the position of the user
+	 * @returns {number} Watch ID
+	 */
+	watchCurrentPosition(onPosChange) {
+		const defaultPosition = L.latLng(defaultLat, defaultLng);
+
+		const options = {
+			enableHighAccuracy: true,
+			timeout: 5000,
+			maximumAge: 0
+		};
+
+		const watchId = navigator.geolocation.watchPosition(
+			position => {
+				const {
+					coords
+				} = position;
+				const {
+					latitude, longitude
+				} = coords;
+
+				onPosChange(L.latLng(latitude, longitude));
+			},
+			() => onPosChange(defaultPosition),
+			options
+		);
+		this.watchId = watchId;
+
+		return watchId;
+	}
+
+	/**
+	 * Clear the current watch
+	 */
+	clearWatchCurrentPosition() {
+		if (!this.watchId) throw new Error('Not currently watching position');
+
+		navigator.geolocation.clearWatch(this.watchId);
+
+		this.watchId = null;
+	}
+}
+
+// Keep these variables outside, so you can use the console inspection tool to debug

```

Comment:
Couldn't you also use the watch tool on mymap / set a debugger?

Reply:
I think so, though I kept it there for other people to easily access the variable in the `Console` tab.